<a href="https://colab.research.google.com/github/teerasitk/signalProcessingInFinance/blob/main/Assignment_04_Solution.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# SET 50 Statistics

1. List of SET50 in to array

##1. Install yahoo fiance

In [ ]:
!pip install yfinance
import fix_yahoo_finance
import yfinance as yf
import pandas as pd
import numpy as np 
import requests 

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 50.4/50.4 KB 2.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.1/4.1 MB 37.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.8/62.8 KB 8.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 112.2/112.2 KB 13.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 128.2/128.2 KB 16.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 111.0/111.0 KB 13.7 MB/s eta 0:00:00
  Attempting uninstall: requests
    Found existing installation: requests 2.25.1
    Uninstalling requests-2.25.1:
      Successfully uninstalled requests-2.25.1
  Attempting uninstall: html5lib
    Found existing installation: html5lib 1.0.1
    Uninstalling html5lib-1.0.1:
      Successfully uninstalled html5lib-1.0.1
  Attempting uninstall: beautifulsoup4
    Found existing installation: beautifulsoup4 4.6.3
   

In [ ]:
url = 'https://classic.set.or.th/mkt/sectorquotation.do?sector=SET50&language=en&country=US'
html = requests.get(url).content
df_list = pd.read_html(html)
df = df_list[-1]
set50_lis = df['Symbol']
set50 = np.array(set50_lis)
print("SET 50 List are: ")
print(set50)


SET 50 List are: 
['ADVANC' 'AOT' 'AWC' 'BANPU' 'BBL' 'BDMS' 'BEM' 'BGRIM' 'BH' 'BTS' 'CBG'
 'CENTEL' 'COM7' 'CPALL' 'CPF' 'CPN' 'CRC' 'DELTA' 'DTAC' 'EA' 'EGCO'
 'GLOBAL' 'GPSC' 'GULF' 'HMPRO' 'INTUCH' 'IVL' 'JMART' 'JMT' 'KBANK' 'KTB'
 'KTC' 'LH' 'MINT' 'MTC' 'OR' 'OSP' 'PTT' 'PTTEP' 'PTTGC' 'RATCH' 'SCB'
 'SCC' 'SCGP' 'TIDLOR' 'TISCO' 'TOP' 'TRUE' 'TTB' 'TU']


##2. Download all the stock prices in 2022 (Use data upto 2022-12-20

In [ ]:
#Example code to download price on one stock in 2021
stock_info = dict() # create empty dictionary
sym = "ADVANC" # stock symbol
df = yf.download(sym + ".BK", start="2022-01-01", end="2022-12-20") # download 2022 data
close_price = df['Adj Close'].dropna() # drop NA out from adjusted closed price
close_price = close_price.values # take out just values
ret = close_price[1:]/close_price[:-1] - 1 # daily return
annual_ret = close_price[-1]/ close_price[0] - 1 # annual return
stock_info[sym] = [close_price[0], close_price[-1],annual_ret, ret.mean(), ret.std()] # record statistics
# make it a pandas dataframe
stock_info = pd.DataFrame(stock_info, index=["Open", "Close", "Annual Return", "Average Daily Return", "Average Daily STD"])
stock_info = stock_info.transpose() # change index to stock symbol
stock_info.to_excel("advance.xlsx") # save to excel file

[*********************100%***********************]  1 of 1 completed


In [ ]:
def stockData(sym):
    stock_info = dict() # create empty dictionary
    # sym = "ADVANC" # stock symbol
    df = yf.download(sym + ".BK", start="2022-01-01", end="2022-12-20") # download 2022 data
    close_price = df['Adj Close'].dropna() # drop NA out from adjusted closed price
    close_price = close_price.values # take out just values
    ret = close_price[1:]/close_price[:-1] - 1 # daily return
    annual_ret = close_price[-1]/ close_price[0] - 1 # annual return
    stock_info[sym] = [close_price[0], close_price[-1],annual_ret, ret.mean(), ret.std()] # record statistics
    stock_info = pd.DataFrame(stock_info, index=["Open", "Close", "Annual Return", "Average Daily Return", "Average Daily STD"])
    stock_info = stock_info.transpose() # change index to stock symbol
    return stock_info

## 3. Download advance.xlsx
What do you see in the file? 



### Answer to Question 3:


1.   Open Price
2.   Close Price
3.  Annual Return
4. Average Daily Return
5. Average Daily STD of 

ADVANC

## 4. Determine Open, Close, Annaul Return, Average Daily Return, and Average Daily STD of all set 50 stocks  and save into set50.*xlsx*

In [ ]:
# Your code is here

s50 = pd.DataFrame([])
for i in set50:
    # print(i)
    df = stockData(i)
    s50 = s50.append(df)
display(s50)
s50.to_excel("set50.xlsx")

[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%********

Open       Close  Annual Return  Average Daily Return  \
ADVANC  220.108490  186.000000      -0.154962             -0.000657   
AOT      61.750000   73.750000       0.194332              0.000812   
AWC       4.809372    6.200000       0.289149              0.001217   
BANPU    10.039300   13.500000       0.344715              0.001458   
BBL     121.375336  143.500000       0.182283              0.000790   
BDMS     21.994970   28.250000       0.284385              0.001192   
BEM       8.470000    9.650000       0.139315              0.000636   
BGRIM    40.130531   39.500000      -0.015712              0.000106   
BH      138.525436  212.000000       0.530405              0.001989   
BTS       9.127392    8.300000      -0.090649             -0.000340   
CBG     117.063957   96.250000      -0.177800             -0.000724   
CENTEL   31.750000   48.500000       0.527559              0.002028   
COM7     39.893036   31.750000      -0.204122             -0.000766   
CPALL    59.202675   66.000000       0.114814              0.000549   
CPF      26.305990   24.100000      -0.083859             -0.000310   
CPN      55.405312   71.000000       0.281466              0.001191   
CRC      32.251591   45.000000       0.395280              0.001560   
DELTA   398.341461  684.000000       0.717120              0.003118   
DTAC     43.893620   44.500000       0.013815              0.000237   
EA       96.907768   96.000000      -0.009367              0.000100   
EGCO    170.082245  169.500000      -0.003423              0.000039   
GLOBAL   19.019753   20.900000       0.098858              0.000530   
GPSC     86.821182   72.500000      -0.164950             -0.000654   
GULF     47.086140   54.250000       0.152144              0.000732   
HMPRO    14.442390   15.200000       0.052457              0.000303   
INTUCH   75.160767   74.381615      -0.010366              0.000009   
IVL      42.112614   40.000000      -0.050166             -0.000068   
JMART    56.501808   41.000000      -0.274360             -0.001093   
JMT      68.695923   67.000000      -0.024687              0.000159   
KBANK   142.453583  144.000000       0.010856              0.000124   
KTB      13.075714   17.799999       0.361302              0.001427   
KTC      58.983047   59.500000       0.008764              0.000107   
LH        8.422523    9.800000       0.163547              0.000724   
MINT     29.750000   31.250000       0.050420              0.000360   
MTC      58.530537   38.250000      -0.346495             -0.001616   
OR       26.702080   23.400000      -0.123664             -0.000488   
OSP      33.379391   28.500000      -0.146180             -0.000583   
PTT      35.918053   31.250000      -0.129964             -0.000532   
PTTEP   113.672440  166.000000       0.460336              0.001799   
PTTGC    56.677677   46.500000      -0.179571             -0.000754   
RATCH    43.187977   42.500000      -0.015930              0.000008   
SCB      80.318840  103.500000       0.288614              0.002166   
SCC     370.767181  331.000000      -0.107256             -0.000445   
SCGP     67.984047   56.000000      -0.176277             -0.000726   
TIDLOR   34.784912   29.000000      -0.166305             -0.000627   
TISCO    91.421860   98.750000       0.080157              0.000354   
TOP      46.460571   53.750000       0.156895              0.000791   
TRUE      4.653387    4.780000       0.027209              0.000306   
TTB       1.405333    1.350000      -0.039374             -0.000081   
TU       18.848131   17.000000      -0.098054             -0.000336   

        Average Daily STD  
ADVANC           0.011987  
AOT              0.009257  
AWC              0.015313  
BANPU            0.018713  
BBL              0.011373  
BDMS             0.014718  
BEM              0.011923  
BGRIM            0.018715  
BH               0.017118  
BTS              0.011960  
CBG              0.015700  
CENTEL           0.019607  
COM7             0.021138  


## 5. sort stocks from highest annual return to the lowest one. 
Here dataframe.sort_values(ascending=False) to sort largest to smallest annual retur

In [ ]:
# try it by yourselves
s50.sort_values(by=["Annual Return"],ascending=False)

Open       Close  Annual Return  Average Daily Return  \
DELTA   398.341461  684.000000       0.717120              0.003118   
BH      138.525436  212.000000       0.530405              0.001989   
CENTEL   31.750000   48.500000       0.527559              0.002028   
PTTEP   113.672440  166.000000       0.460336              0.001799   
CRC      32.251591   45.000000       0.395280              0.001560   
KTB      13.075714   17.799999       0.361302              0.001427   
BANPU    10.039300   13.500000       0.344715              0.001458   
AWC       4.809372    6.200000       0.289149              0.001217   
SCB      80.318840  103.500000       0.288614              0.002166   
BDMS     21.994970   28.250000       0.284385              0.001192   
CPN      55.405312   71.000000       0.281466              0.001191   
AOT      61.750000   73.750000       0.194332              0.000812   
BBL     121.375336  143.500000       0.182283              0.000790   
LH        8.422523    9.800000       0.163547              0.000724   
TOP      46.460571   53.750000       0.156895              0.000791   
GULF     47.086140   54.250000       0.152144              0.000732   
BEM       8.470000    9.650000       0.139315              0.000636   
CPALL    59.202675   66.000000       0.114814              0.000549   
GLOBAL   19.019753   20.900000       0.098858              0.000530   
TISCO    91.421860   98.750000       0.080157              0.000354   
HMPRO    14.442390   15.200000       0.052457              0.000303   
MINT     29.750000   31.250000       0.050420              0.000360   
TRUE      4.653387    4.780000       0.027209              0.000306   
DTAC     43.893620   44.500000       0.013815              0.000237   
KBANK   142.453583  144.000000       0.010856              0.000124   
KTC      58.983047   59.500000       0.008764              0.000107   
EGCO    170.082245  169.500000      -0.003423              0.000039   
EA       96.907768   96.000000      -0.009367              0.000100   
INTUCH   75.160767   74.381615      -0.010366              0.000009   
BGRIM    40.130531   39.500000      -0.015712              0.000106   
RATCH    43.187977   42.500000      -0.015930              0.000008   
JMT      68.695923   67.000000      -0.024687              0.000159   
TTB       1.405333    1.350000      -0.039374             -0.000081   
IVL      42.112614   40.000000      -0.050166             -0.000068   
CPF      26.305990   24.100000      -0.083859             -0.000310   
BTS       9.127392    8.300000      -0.090649             -0.000340   
TU       18.848131   17.000000      -0.098054             -0.000336   
SCC     370.767181  331.000000      -0.107256             -0.000445   
OR       26.702080   23.400000      -0.123664             -0.000488   
PTT      35.918053   31.250000      -0.129964             -0.000532   
OSP      33.379391   28.500000      -0.146180             -0.000583   
ADVANC  220.108490  186.000000      -0.154962             -0.000657   
GPSC     86.821182   72.500000      -0.164950             -0.000654   
TIDLOR   34.784912   29.000000      -0.166305             -0.000627   
SCGP     67.984047   56.000000      -0.176277             -0.000726   
CBG     117.063957   96.250000      -0.177800             -0.000724   
PTTGC    56.677677   46.500000      -0.179571             -0.000754   
COM7     39.893036   31.750000      -0.204122             -0.000766   
JMART    56.501808   41.000000      -0.274360             -0.001093   
MTC      58.530537   38.250000      -0.346495             -0.001616   

        Average Daily STD  
DELTA            0.040231  
BH               0.017118  
CENTEL           0.019607  
PTTEP            0.017888  
CRC              0.015366  
KTB              0.013597  
BANPU            0.018713  
AWC              0.015313  
SCB              0.035855  
BDMS             0.014718  
CPN              0.015322  
AOT              0.009257  
BBL              0.011373  


### Answer here
Stock with Higest Return: __*DELTA*____ and Stock with Lowest Return: __*MTC*____

## 6. Sort stocks based on the standard deviation from highest to lowest one. 

In [ ]:
# try it by yourselves

s50.sort_values(by=["Average Daily STD"],ascending=False)

Open       Close  Annual Return  Average Daily Return  \
DELTA   398.341461  684.000000       0.717120              0.003118   
SCB      80.318840  103.500000       0.288614              0.002166   
JMART    56.501808   41.000000      -0.274360             -0.001093   
JMT      68.695923   67.000000      -0.024687              0.000159   
COM7     39.893036   31.750000      -0.204122             -0.000766   
MTC      58.530537   38.250000      -0.346495             -0.001616   
CENTEL   31.750000   48.500000       0.527559              0.002028   
TRUE      4.653387    4.780000       0.027209              0.000306   
DTAC     43.893620   44.500000       0.013815              0.000237   
BGRIM    40.130531   39.500000      -0.015712              0.000106   
BANPU    10.039300   13.500000       0.344715              0.001458   
TIDLOR   34.784912   29.000000      -0.166305             -0.000627   
PTTEP   113.672440  166.000000       0.460336              0.001799   
TOP      46.460571   53.750000       0.156895              0.000791   
IVL      42.112614   40.000000      -0.050166             -0.000068   
MINT     29.750000   31.250000       0.050420              0.000360   
BH      138.525436  212.000000       0.530405              0.001989   
EA       96.907768   96.000000      -0.009367              0.000100   
GPSC     86.821182   72.500000      -0.164950             -0.000654   
CBG     117.063957   96.250000      -0.177800             -0.000724   
GLOBAL   19.019753   20.900000       0.098858              0.000530   
GULF     47.086140   54.250000       0.152144              0.000732   
CRC      32.251591   45.000000       0.395280              0.001560   
CPN      55.405312   71.000000       0.281466              0.001191   
AWC       4.809372    6.200000       0.289149              0.001217   
SCGP     67.984047   56.000000      -0.176277             -0.000726   
TU       18.848131   17.000000      -0.098054             -0.000336   
BDMS     21.994970   28.250000       0.284385              0.001192   
PTTGC    56.677677   46.500000      -0.179571             -0.000754   
OSP      33.379391   28.500000      -0.146180             -0.000583   
TTB       1.405333    1.350000      -0.039374             -0.000081   
KTB      13.075714   17.799999       0.361302              0.001427   
OR       26.702080   23.400000      -0.123664             -0.000488   
HMPRO    14.442390   15.200000       0.052457              0.000303   
CPALL    59.202675   66.000000       0.114814              0.000549   
KBANK   142.453583  144.000000       0.010856              0.000124   
RATCH    43.187977   42.500000      -0.015930              0.000008   
ADVANC  220.108490  186.000000      -0.154962             -0.000657   
BTS       9.127392    8.300000      -0.090649             -0.000340   
BEM       8.470000    9.650000       0.139315              0.000636   
PTT      35.918053   31.250000      -0.129964             -0.000532   
CPF      26.305990   24.100000      -0.083859             -0.000310   
KTC      58.983047   59.500000       0.008764              0.000107   
LH        8.422523    9.800000       0.163547              0.000724   
BBL     121.375336  143.500000       0.182283              0.000790   
EGCO    170.082245  169.500000      -0.003423              0.000039   
INTUCH   75.160767   74.381615      -0.010366              0.000009   
SCC     370.767181  331.000000      -0.107256             -0.000445   
AOT      61.750000   73.750000       0.194332              0.000812   
TISCO    91.421860   98.750000       0.080157              0.000354   

        Average Daily STD  
DELTA            0.040231  
SCB              0.035855  
JMART            0.024213  
JMT              0.023183  
COM7             0.021138  
MTC              0.021112  
CENTEL           0.019607  
TRUE             0.019461  
DTAC             0.018791  
BGRIM            0.018715  
BANPU            0.018713  
TIDLOR           0.017918  
PTTEP            0.017888  


### Answer here
Stock with Higest STD:_*DELTA*____ and Stock with Lowest Return: __*TISCO*__

## 7. Sort a stock based on the ratio of average daily return over average daily std 

In [ ]:
# try it yourselves

s50["Ratio"] = s50["Average Daily Return"]/s50["Average Daily STD"]
s50.sort_values(by=["Ratio"],ascending=False)

Open       Close  Annual Return  Average Daily Return  \
BH      138.525436  212.000000       0.530405              0.001989   
KTB      13.075714   17.799999       0.361302              0.001427   
CENTEL   31.750000   48.500000       0.527559              0.002028   
CRC      32.251591   45.000000       0.395280              0.001560   
PTTEP   113.672440  166.000000       0.460336              0.001799   
AOT      61.750000   73.750000       0.194332              0.000812   
BDMS     21.994970   28.250000       0.284385              0.001192   
AWC       4.809372    6.200000       0.289149              0.001217   
BANPU    10.039300   13.500000       0.344715              0.001458   
CPN      55.405312   71.000000       0.281466              0.001191   
DELTA   398.341461  684.000000       0.717120              0.003118   
BBL     121.375336  143.500000       0.182283              0.000790   
LH        8.422523    9.800000       0.163547              0.000724   
SCB      80.318840  103.500000       0.288614              0.002166   
TISCO    91.421860   98.750000       0.080157              0.000354   
BEM       8.470000    9.650000       0.139315              0.000636   
GULF     47.086140   54.250000       0.152144              0.000732   
TOP      46.460571   53.750000       0.156895              0.000791   
CPALL    59.202675   66.000000       0.114814              0.000549   
GLOBAL   19.019753   20.900000       0.098858              0.000530   
HMPRO    14.442390   15.200000       0.052457              0.000303   
MINT     29.750000   31.250000       0.050420              0.000360   
TRUE      4.653387    4.780000       0.027209              0.000306   
DTAC     43.893620   44.500000       0.013815              0.000237   
KBANK   142.453583  144.000000       0.010856              0.000124   
KTC      58.983047   59.500000       0.008764              0.000107   
JMT      68.695923   67.000000      -0.024687              0.000159   
EA       96.907768   96.000000      -0.009367              0.000100   
BGRIM    40.130531   39.500000      -0.015712              0.000106   
EGCO    170.082245  169.500000      -0.003423              0.000039   
INTUCH   75.160767   74.381615      -0.010366              0.000009   
RATCH    43.187977   42.500000      -0.015930              0.000008   
IVL      42.112614   40.000000      -0.050166             -0.000068   
TTB       1.405333    1.350000      -0.039374             -0.000081   
TU       18.848131   17.000000      -0.098054             -0.000336   
CPF      26.305990   24.100000      -0.083859             -0.000310   
BTS       9.127392    8.300000      -0.090649             -0.000340   
TIDLOR   34.784912   29.000000      -0.166305             -0.000627   
COM7     39.893036   31.750000      -0.204122             -0.000766   
OR       26.702080   23.400000      -0.123664             -0.000488   
OSP      33.379391   28.500000      -0.146180             -0.000583   
GPSC     86.821182   72.500000      -0.164950             -0.000654   
PTT      35.918053   31.250000      -0.129964             -0.000532   
JMART    56.501808   41.000000      -0.274360             -0.001093   
CBG     117.063957   96.250000      -0.177800             -0.000724   
SCC     370.767181  331.000000      -0.107256             -0.000445   
SCGP     67.984047   56.000000      -0.176277             -0.000726   
PTTGC    56.677677   46.500000      -0.179571             -0.000754   
ADVANC  220.108490  186.000000      -0.154962             -0.000657   
MTC      58.530537   38.250000      -0.346495             -0.001616   

        Average Daily STD     Ratio  
BH               0.017118  0.116206  
KTB              0.013597  0.104963  
CENTEL           0.019607  0.103416  
CRC              0.015366  0.101519  
PTTEP            0.017888  0.100594  
AOT              0.009257  0.087703  
BDMS             0.014718  0.081007  
AWC              0.015313  0.079473  
BANPU            0.018713  0.077899  
CPN         

### Answer here
Stock with Higest Ratio:**_*BH*_____** and Stock with Lowest Return: **__*MTC*___**

## 8. Find the mean and standard deviation ratio of all stocks.

Note that you will use ".mean()" and ".std()"

In [ ]:
# try it yourselves
ar_mean = s50["Annual Return"].mean()
ar_std = s50["Annual Return"].std()
ar_mean,ar_std

(0.06265406853457692, 0.2271825753594581)

Mean: **___*0.06265*____** and STD:**_*0.22718*___**

## 9. We will use z-score to see each stock performance 
$z = \frac{r - \mu_r}{\sigma_r}$


where $\mu_r$, and $\sigma_r$ are mean, snd standard deviation of stocks.

Compute z score from all stocks

In [ ]:
# try it yourselves

z_socre = (s50["Annual Return"]-ar_mean)/ar_std
display(z_socre)

ADVANC   -0.957891
AOT       0.579613
AWC       0.996975
BANPU     1.241562
BBL       0.526576
BDMS      0.976001
BEM       0.337443
BGRIM    -0.344948
BH        2.058920
BTS      -0.674802
CBG      -1.058417
CENTEL    2.046394
COM7     -1.174279
CPALL     0.229597
CPF      -0.644913
CPN       0.963153
CRC       1.464135
DELTA     2.880792
DTAC     -0.214978
EA       -0.317020
EGCO     -0.290856
GLOBAL    0.159359
GPSC     -1.001857
GULF      0.393911
HMPRO    -0.044883
INTUCH   -0.321418
IVL      -0.496604
JMART    -1.483448
JMT      -0.384455
KBANK    -0.228004
KTB       1.314574
KTC      -0.237209
LH        0.444104
MINT     -0.053851
MTC      -1.800970
OR       -0.820124
OSP      -0.919233
PTT      -0.847856
PTTEP     1.750497
PTTGC    -1.066214
RATCH    -0.345906
SCB       0.994619
SCC      -0.747903
SCGP     -1.051715
TIDLOR   -1.007821
TISCO     0.077045
TOP       0.414824
TRUE     -0.156021
TTB      -0.449101
TU       -0.707395
Name: Annual Return, dtype: float64

## 10. If we wan to test between
$H_0$: Stock perform similarly to the group of SET50

$H_1$: Stock perform better than the group of SET50

$p$-value is compute from 
$p= 1-\Phi(z)$

**Determine stocks with $p$-value less than 0.05, namely stocks that outperforms the group.**

In [ ]:
# example code for compute p-value for z = 1
from scipy.stats import norm

p =  1- norm.cdf(z_socre)
p_value = pd.DataFrame(p,index=list(s50.index),columns=["P-value"])
print(p_value)

         P-value
ADVANC  0.830941
AOT     0.281088
AWC     0.159388
BANPU   0.107199
BBL     0.299244
BDMS    0.164532
BEM     0.367892
BGRIM   0.634933
BH      0.019751
BTS     0.750099
CBG     0.855067
CENTEL  0.020359
COM7    0.879858
CPALL   0.409202
CPF     0.740508
CPN     0.167735
CRC     0.071579
DELTA   0.001983
DTAC    0.585108
EA      0.624386
EGCO    0.614419
GLOBAL  0.436693
GPSC    0.841794
GULF    0.346823
HMPRO   0.517900
INTUCH  0.626053
IVL     0.690266
JMART   0.931022
JMT     0.649679
KBANK   0.590178
KTB     0.094327
KTC     0.593752
LH      0.328484
MINT    0.521473
MTC     0.964146
OR      0.793927
OSP     0.821013
PTT     0.801741
PTTEP   0.040016
PTTGC   0.856837
RATCH   0.635293
SCB     0.159961
SCC     0.772741
SCGP    0.853535
TIDLOR  0.843230
TISCO   0.469294
TOP     0.339135
TRUE    0.561992
TTB     0.673321
TU      0.760340


In [ ]:

p_value[p_value["P-value"]<0.05]

P-value
BH      0.019751
CENTEL  0.020359
DELTA   0.001983
PTTEP   0.040016

### Answer: **__*BH, CENTEL, DELTA, PTTEP*___** outperforms the rest of SET50 with significance level of 5%
